src/main/java/com/atlassian/httpclient/apache/httpcomponents/SettableFuturePromiseHttpPromiseAsyncClient.java

```diff
@@ -39,24 +44,38 @@
             void doCompleted(final HttpResponse httpResponse)
             {
                 executor.execute( () -> future.set(httpResponse));
+                log.info( "Closing in doCompleted()" );
+                closeClient();
             }
 
             @Override
             void doFailed(final Exception ex)
             {
                 executor.execute(() -> future.setException(ex));
+                log.info( "Closing in doFailed()" );
+                closeClient();
             }
 
             @Override
             void doCancelled()
             {
                 final TimeoutException timeoutException = new TimeoutException();
                 executor.execute(() -> future.setException(timeoutException));
+                log.info( "Closing in doCancelled()" );
+                closeClient();
             }
         });
         return Promises.forListenableFuture(future);
     }
 
+    private void closeClient() {
+        try {
+            client.close();
+        } catch ( IOException e ) {
+            log.error( "Close failed" , e );
```

Comment:
How about making it a bit more meaningful, like `("Failed to close HttpAsyncClient", e);` ?